python_ta/checkers/invalid_name_checker.py

```diff
@@ -0,0 +1,525 @@
+    """Returns a list of strings, each detailing how `name` violates the PEP8 naming conventions for
+    type alias names.
+
+    Returns an empty list if `name` is a valid type alias name."""
+    error_msgs = []
+
+    bad_name_msg = _is_bad_name(name)
+    if bad_name_msg:
+        error_msgs.append(bad_name_msg)
+
+    if not _is_in_pascal_case(name):
+        error_msgs.append(
+            f'Typealias name "{name}" is not in PascalCase. Typealias names should '
+            f"have the first letter of each word capitalized with no separation "
+            f"between each word."
+        )
+
+    return error_msgs
+
+
+# Map each variable name type to its corresponding check
+NAME_CHECK = {
+    "module": _check_module_name,
+    "const": _check_const_name,
+    "class": _check_class_name,
+    "function": _check_function_and_variable_name,
+    "method": _check_method_and_attr_name,
+    "attr": _check_method_and_attr_name,
+    "argument": _check_argument_name,
+    "variable": _check_function_and_variable_name,
+    "class_attribute": _check_method_and_attr_name,
+    "class_const": _check_const_name,
+    "inlinevar": _check_inlinevar_name,
+    "typevar": _check_typevar_name,
+    "typealias": _check_typealias_name,
+}
+
+
+class InvalidNameChecker(BaseChecker):
+    """A checker class to report on names that don't conform to PEP8 naming conventions.
+
+    For the PEP8 naming conventions, see https://peps.python.org/pep-0008/#naming-conventions.
+    """
+
+    name = "pep8_name_violation"
+    # Template of displayed message will include the following:
+    #   1. Describe the variable name type (e.g. `module`, `const`, etc.) and the name itself.
+    #   2. Explanation of how the name violates the naming convention and suggest a
+    #   correction(s), if appropriate.
+    msgs = {
+        "C9103": (
+            'Found %s name "%s" in violation of PEP8 naming conventions. %s',
+            "pep8-name-violation",
+            "Used when the name doesn't conform to naming conventions specified "
+            "in the PEP8 style guide.",
+        ),
+    }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @only_required_for_messages("pep8-name-violation")
+    def visit_module(self, node: nodes.Module) -> None:
+        """Visit a Module node to check for any name violations.
+
+        Snippets taken from pylint.checkers.base.name_checker.checker."""
+        self._check_name("module", node.name.split(".")[-1], node)
+
+    @only_required_for_messages("pep8-name-violation")
+    def visit_classdef(self, node: nodes.ClassDef):
+        """Visit a Class node to check for any name violations.
+
+        Taken from pylint.checkers.base.name_checker.checker."""
+        self._check_name("class", node.name, node)
+        # Check any instance attributes that were first defined in this class.
+        for attr, anodes in node.instance_attrs.items():
```

Comment:
Don't check this here. Instead, check only when instance attributes are "declared", which by our convention is in the class body as type annotations.